### PR TITLE
UCT/DC: Use user_index to identify QP

### PR DESCRIPTION
**Pull request by Artemy-Mellanox**

### CODE REVIEW

**Comment by Artemy-Mellanox** [comment]
bot:pipe:retest

File: src/uct/ib/dc/dc_mlx5.c [hunk 1]
```diff
@@ -218,8 +218,12 @@ uct_dc_mlx5_poll_tx(uct_dc_mlx5_iface_t *iface)
 
     ucs_memory_cpu_load_fence();
 
-    qp_num = ntohl(cqe->sop_drop_qpn) & UCS_MASK(UCT_IB_QPN_ORDER);
-    dci = uct_dc_mlx5_iface_dci_find(iface, qp_num);
+    if (iface->flags & UCT_DC_MLX5_IFACE_FLAG_UIDX) {
+        dci = ntohl(cqe->srqn_uidx) & UCS_MASK(UCT_IB_QPN_ORDER);
+    } else {
+        qp_num = ntohl(cqe->sop_drop_qpn) & UCS_MASK(UCT_IB_QPN_ORDER);
+        dci = uct_dc_mlx5_iface_dci_find(iface, qp_num);
```
Comment (yosefe) [inline src/uct/ib/dc/dc_mlx5.c:225]: align =

File: src/uct/ib/dc/dc_mlx5.c [hunk 2]
```diff
@@ -218,8 +218,12 @@ uct_dc_mlx5_poll_tx(uct_dc_mlx5_iface_t *iface)
 
     ucs_memory_cpu_load_fence();
 
-    qp_num = ntohl(cqe->sop_drop_qpn) & UCS_MASK(UCT_IB_QPN_ORDER);
-    dci = uct_dc_mlx5_iface_dci_find(iface, qp_num);
+    if (iface->flags & UCT_DC_MLX5_IFACE_FLAG_UIDX) {
+        dci = ntohl(cqe->srqn_uidx) & UCS_MASK(UCT_IB_QPN_ORDER);
```
Comment (yosefe) [inline src/uct/ib/dc/dc_mlx5.c:222]: can we avoid ntohl here?

File: src/uct/ib/dc/dc_mlx5.c [hunk 3]
```diff
@@ -294,7 +294,7 @@ static ucs_status_t uct_dc_mlx5_iface_create_dci(uct_dc_mlx5_iface_t *iface,
 
     if (md->flags & UCT_IB_MLX5_MD_FLAG_DEVX_DCI) {
         attr.mmio_mode                        = iface->super.tx.mmio_mode;
-        attr.uidx                             = ndci;
+        attr.uidx                             = htonl(ndci) >> UCT_IB_UIDX_SHIFT;
```
Comment (yosefe) [inline src/uct/ib/dc/dc_mlx5.c:297]: 👍

**Comment by Artemy-Mellanox** [comment]
bot:pipe:retest

**Review by yosefe — APPROVED** [review]
pls squash

**Pull request merged**